src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -382,6 +380,34 @@ public static boolean deleteDirectory(File path) {
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from remote origin HEAD.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
+     */
+    public static String getDefaultBranchName(final String repositoryUrl) {
+        try {
+            Map<String, Ref> gitMap = Git.lsRemoteRepository().setRemote(repositoryUrl).callAsMap();
+            Ref head = gitMap.get("HEAD");
+            if (head.isSymbolic()) {
+                return getBranchFromRef(head.getTarget().getName());
+            } else {
+                for (String refname : gitMap.keySet()) {
+                    if (refname.startsWith("refs/heads/")
+                            && head.getObjectId().equals(gitMap.get(refname).getObjectId())) {
+                        return getBranchFromRef(refname);
+                    }
+                }
+            }
+        } catch (NullPointerException | GitAPIException ignore) {
```

Comment:
I don't really like recovering from NPEs by try-catch. I would prefer an explicit null check with early return.

Since the early return would be also returning `"master"`, you could add a constant `DEFAULT_LOCAL_BRANCH` or something.